src/Hateoas/HateoasBuilder.php

```diff
@@ -183,9 +183,12 @@ public function build()
         ;
 
         $jmsSerializer = $this->serializerBuilder->build();
+        $ref = new \ReflectionProperty(get_class($jmsSerializer), 'factory');
```

Comment:
Why was the getter removed ?

Reply:
`getMetadataFactory` has been removed in jms serializer to keep the serializer api simpler and cleaner.

In general getting dependencies should be obtained via some DI container (as it is done in the symfony bundles) instead of getting them from the serializer object.

IMO, if we want to use the builders, should make them able to provide the right metadata factory.